### PR TITLE
Add Crab Issuing Module

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

File: bin/node-template/runtime/src/lib.rs [hunk 1]
```diff
@@ -812,7 +812,17 @@ impl pallet_sudo::Trait for Runtime {
...
+}
+impl darwinia_crab_issuing::Trait for Runtime {
+	type Event = Event;
+	type ModuleId = CrabIssuingModuleId;
+	type RingCurrency = Ring;
+	type WeightInfo = ();
+}
+
+parameter_types! {
+	pub const EthBackingModuleId: ModuleId = ModuleId(*b"da/ethbk");
```
Comment (HackFisher) [inline bin/node-template/runtime/src/lib.rs:825]: Changing the EthBackingModuleId could cause related account address being changes.

Related account address might have tokens in genesis.

We need to print these module account id and do some check on Crab.

File: frame/bridge/crab/issuing/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+impl WeightInfo for () {}
+
+decl_event! {
+	pub enum Event<T>
+	where
+		AccountId = AccountId<T>,
+		RingBalance = RingBalance<T>,
+	{
+		/// Someone swapped some *CRING*. [who, swapped *CRING*, burned Mapped *RING*]
+		SwapAndBurn(AccountId, RingBalance, MappedRing),
```
Comment (HackFisher) [inline frame/bridge/crab/issuing/src/lib.rs:54]: Change to SwapAndBurnToGensis

File: frame/bridge/crab/issuing/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+			/// Swap Amount - TOO LOW
+			SwapAmountTL,
+			/// Backed *RING* - INSUFFICIENT
+			BackedRingIS,
+	}
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as DarwiniaCrabIssuing {
+		pub BackedRing
```
Comment (HackFisher) [inline frame/bridge/crab/issuing/src/lib.rs:69]: BackedRing might have concept confusion
```suggestion
		pub TotalMappedRing
```

File: frame/bridge/crab/issuing/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	pub struct Module<T: Trait> for enum Call
+	where
+		origin: T::Origin
+	{
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		#[weight = T::DbWeight::get().reads_writes(2, 1) + 100_000_000]
+		pub fn swap_and_burn(origin, amount: RingBalance<T>) {
```
Comment (HackFisher) [inline frame/bridge/crab/issuing/src/lib.rs:96]:
```suggestion
		pub fn swap_and_burn_to_genesis(origin, amount: RingBalance<T>) {
```

File: frame/bridge/ethereum/backing/src/lib.rs [hunk 5]
```diff
@@ -100,6 +100,7 @@ mod types {
 
 	pub type RingBalance<T> =
 		<<T as Trait>::RingCurrency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+	#[cfg(feature = "std")]
```
Comment (HackFisher) [inline frame/bridge/ethereum/backing/src/lib.rs:103]: Why need to add this?

**Review by HackFisher — CHANGES_REQUESTED** [review]

File: bin/node-template/runtime/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply bin/node-template/runtime/src/lib.rs:825]: Yes.

File: frame/bridge/ethereum/backing/src/lib.rs (hunk 5 above)
Comment (AurevoirXavier) [reply frame/bridge/ethereum/backing/src/lib.rs:103]: Just suppress the `unused import` warning in no std.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**